02_classify_stratified/nbconverted/plot_purity_regression_results.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Plot tumor purity prediction results
+# 
+# Here, we'll visualize the results of our tumor purity prediction. We're predicting real-valued purity labels here (e.g. elastic net regression).
```

Comment:
I know ElasticNet is pretty standard, but any reason why you chose this one in particular? Maybe a biological reason? How did you specify the regularization parameters?

Reply:
Right, good question. A few observations about this:
* All of these datasets are high-dimensional (gene expression and methylation especially) and only a few of the features (genes or methylation probes) are likely to be relevant. Elastic net tends to be a good way to balance performance and feature selection for problems like this, and in similar past work it's tended to work well.
* If you're asking specifically about the sklearn `ElasticNet` vs. other ways of fitting a regularized regression model, I found it to be a good balance between runtime and accuracy. I also tried sklearn `SGDRegressor` and found it to be much more finicky/sensitive to parameter choices. `ElasticNet` uses coordinate descent rather than batch SGD to fit the model.
* For the parameters, I specify ranges to search over [here](https://github.com/greenelab/mpmp/blob/d2d805c82f15952d47f6c37bcfc704bc4d3c4258/mpmp/config.py#L85), then I'm doing a grid search on the training dataset to choose the best ones [here](https://github.com/greenelab/mpmp/blob/d2d805c82f15952d47f6c37bcfc704bc4d3c4258/mpmp/prediction/regression.py#L60). I chose the ranges based on [this sklearn documentation](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.ElasticNetCV.html), which suggests values close to 1 for `l1_ratio` and evenly spaced values for `alpha`. Other than that, I didn't do much tuning by hand.

I'll add a bit of documentation about the parameter ranges in `regression.py`, just for future reference.